### PR TITLE
[EJBCLIENT-178] Fix EJB JNDI parsing to be able to handle the case where an empty distinct name is not included in the JNDI name

### PR DESCRIPTION
**Pull request by fjuma**
https://issues.jboss.org/browse/EJBCLIENT-178

### CODE REVIEW

File: src/main/java/org/jboss/ejb/client/EJBRootContext.java [hunk 1]
```diff
@@ -83,9 +83,15 @@ protected Object lookupNative(final Name name) throws NamingException {
             }
         }
         if (beanName == null) {
+            if (size == 3) {
+                // name is of the form appName/moduleName/distinctName
+                return new RelativeContext(new FastHashtable<>(getEnvironment()), this, SimpleName.of(name));
+            }
```
Comment (dmlloyd) [inline src/main/java/org/jboss/ejb/client/EJBRootContext.java]: I think that this suffers from a problem where you cannot disambiguate between app/module/distinct vs app/module/bean, doesn't it?

Comment (fjuma) [inline src/main/java/org/jboss/ejb/client/EJBRootContext.java]: The Javadoc in the legacy EjbJndiNameParser indicates that a name of the form a/b/c will be considered to be a partial app/module/distinct lookup as opposed to a lookup of a bean name without a view. See https://github.com/jbossas/jboss-ejb-client/blob/3.x/src/main/java/org/jboss/ejb/client/naming/ejb/EjbJndiNameParser.java#L40-L42.

Comment (dmlloyd) [inline src/main/java/org/jboss/ejb/client/EJBRootContext.java]: I'm trying to work out now if there's some case where the bean name is null where it wasn't meant to be; I guess that must be what's happening?

Comment (dmlloyd) [inline src/main/java/org/jboss/ejb/client/EJBRootContext.java]: Ah I was interpreting you backwards: I read that we did *not* want to support three segments under any circumstances.  But if this was the legacy behavior then we definitely do need to preserve it.

**Pull request merged**